File: code/innobyte/group-49/Set.php
<?php
// Set Data Structure Task for Group 49
// Name: Precious Eyo
// RegNo: 20/csc/181
/**
 * Class set
 * Represents a set data structure.
 */
class SetDS
{
    private $elements;

    /**
     * Constructor to initialize the set.
     */
    public function __construct()
    {
        $this->elements = [];
    }

    /**
     * Add an element to the set.
     * @param mixed $element The element to add.
     */
    public function add($element)
    {
        if (!$this->contains($element)) {
            $this->elements[] = $element;
        }
    }

    /**
     * Remove an element from the set.
     * @param mixed $element The element to remove.
     */
    public function remove($element)
    {
        $index = array_search($element, $this->elements);
        if ($index !== false) {
            unset($this->elements[$index]);
        }
    }

    /**
     * Check if the set contains a specific element.
     * @param mixed $element The element to check.
     * @return bool True if the element is in the set, false otherwise.
     */
    public function contains($element)
    {
        return in_array($element, $this->elements);
    }

    /**
     * Get the size of the set.
     * @return int The number of elements in the set.
     */
    public function size()
    {
        return count($this->elements);
    }

    /**
     * Get all elements in the set.
     * @return array An array containing all elements in the set.
     */
    public function toArray()
    {
        return $this->elements;
    }
}

File: code/innobyte/group-52/Set.py
# Set Data Strucuture tasks assigned to me - group 52
# Author: Isaac Cletus Unimnake
# RegNo: 20/CSC/030

class Set:
    def __init__(self):
        """Initialize the set."""
        self.elements = []
        
    def add(self, element):
        """Add an element to the set."""
        if element not in self.elements:
            self.elements.append(element)

    def remove(self, element):
        """Remove an element from the set."""
        if element in self.elements:
            self.elements.remove(element)
            
    def contains(self, element):
        """Check if the set contains a specific element."""
        return element in self.elements

    def size(self):
        """Get the size of the set."""
        return len(self.elements)

    def to_list(self):
        """Get all elements in the set as a list."""
        return self.elements

#this section is the example usage part for the above structure:
if __name__ == "__main__":
    # Create a new set
    my_set = Set()

    # Add elements to the set
    my_set.add(1)
    my_set.add(2)
    my_set.add(3)

    # Remove an element from the set
    my_set.remove(2)

    # Check if the set contains a specific element
    print("Set contains 1:", my_set.contains(1))  # True
    print("Set contains 2:", my_set.contains(2))  # False

    # Get the size of the set
    print("Size of set:", my_set.size())  # 2

    # Get all elements in the set
    print("Elements in the set:", my_set.to_list())  # [1, 3]
